x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/fielddata/GeometryDocValueReader.java

```diff
@@ -109,4 +112,17 @@ public void visit(TriangleTreeVisitor visitor) throws IOException {
         }
     }
 
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        input.writeTo(out);
+    }
+
+    private static class GeometryByteArrayReader extends ByteArrayStreamInput implements Writeable {
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeBytesReference(new BytesArray(bytes));
```

Comment:
I'm suspicious about this in the same way as I was about the resetting to position 0: `bytes` came from a `BytesRef` which represents a slice of the overall array, so it may have a nonzero `offset` (excluding some bytes at the start) and `offset + length` may be less than the total array length (excluding some bytes at the end). I don't think it's correct to write out the whole array here.

Reply:
Now I'm re-using the incoming ByteRef as @iverase suggested, so hopefully that resolves this concern.